var/spack/repos/builtin/packages/nlcglib/package.py

```diff
@@ -51,18 +63,18 @@ class Nlcglib(CMakePackage, CudaPackage, ROCmPackage):
         depends_on("rocblas")
         depends_on("rocsolver")
 
-    for arch in CudaPackage.cuda_arch_values:
-        depends_on(
-            f"kokkos+cuda+cuda_lambda+wrapper cuda_arch={arch}",
-            when=f"%gcc +cuda cuda_arch={arch}",
-        )
-        depends_on(f"kokkos+cuda cuda_arch={arch}", when=f"+cuda cuda_arch={arch}")
+    with when("+cuda"):
+        depends_on("kokkos+cuda+cuda_lambda+wrapper", when="%gcc")
+        depends_on("kokkos+cuda")
```

Comment:
Sufficient:
```suggestion
        depends_on("kokkos+cuda_lambda+wrapper", when="%gcc")
        depends_on("kokkos+cuda")
```
?
Not a blocker.